### ISSUE TITLE
not possible to enable admin_api_require_auth with helm chart

### ISSUE THREAD

**Issue by brokenjacobs** [redpanda-data/helm-charts#1019]
### What happened?

Initially deploying the helm chart with admin_api_require_auth in the config.cluster settings results in a cluster without this setting enabled. When applying the helm chart again during an upgrade the post upgrade job fails in different ways depending on how you provisiion your superusers. If you provision superusers using a secret and setting the users list to [] you get the following output from the upgrade job:
```
+ kubectl logs -f redpanda-post-upgrade-l2mvq
PROPERTY                    PRIOR                                                   NEW
admin_api_require_auth      false                                                   true
log_message_timestamp_type  CreateTime                                              LogAppendTime
superusers                  [ANONYMOUS superuser]

Validation errors:
 * admin_api_require_auth: No superusers defined
 * 
 ```

If instead you use helm values to set the users list, when you do the second apply you get the following failure:
```
❯ klo redpanda-post-upgrade-qxmj4
+ kubectl logs -f redpanda-post-upgrade-qxmj4
PROPERTY                    PRIOR                                                   NEW
log_message_timestamp_type  CreateTime                                              LogAppendTime
superusers                  [ANONYMOUS superuser]  [superuser ANONYMOUS]
admin_api_require_auth      false                                                   true

Validation errors:
 * admin_api_require_auth: May only be set by a superuser

No changes were made
```

Either way, you get a broken cluster, and you don't get your admin api requiring authentication.

The broker logs on first startup show:
```
INFO  2024-02-08 20:51:10,832 [shard 0:main] cluster - config_manager.cc:425 - Ignoring value for 'admin_api_require_auth' in redpanda.yaml: use `rpk cluster config edit` to edit cluster configuration properties.
```
and then a bit later conveniently:
```
WARN  2024-02-08 20:51:11,069 [shard 0:main] admin_api_server - server.cc:559 - Insecure Admin API listener on 0.0.0.0:9645, consider enabling `admin_api_require_auth
```

You know.. I never considered that, what a wonderful idea!

### What did you expect to happen?

The admin api to be provisioned by the helm chart requiring basic auth authentication.

### How can we reproduce it (as minimally and precisely as possible)?. Please include values file.

```yaml
auth:
  sasl:
    enabled: true
    mechanism: SCRAM-SHA-512
    users:
      - name: ANONYMOUS
        password: ""
        mechanism: SCRAM-SHA-512
      - name: superuser
        password: superduper
        mechanism: SCRAM-SHA-512
config:
  cluster:
    admin_api_require_auth: true
```

This will reproduce the second failure above.

### Anything else we need to know?

TLDR; 
First deploy: admin_api_require_auth doesn't get set
Second deploy: It tries to deploy but breaks.

Breaks differently depending on how you provision your superusers.

### Which are the affected charts?

Redpanda

### Chart Version(s)

5.7.23

### Cloud provider

Locally hosted on k3s.

JIRA Link: [K8S-102](https://redpandadata.atlassian.net/browse/K8S-102)

[K8S-102]: https://redpandadata.atlassian.net/browse/K8S-102?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by JakeSCahill** [comment]
Related issue: https://github.com/redpanda-data/helm-charts/issues/848
